### PR TITLE
Add lint warnings for component prop-types

### PR DESCRIPTION
**Pull request by bafonins**
add warnings for component prop-types

#### Summary

References: https://github.com/TheThingsNetwork/lorawan-stack/issues/1086

#### Changes

- Add eslint rules for prop-types

#### Notes for Reviewers

With this PR we will be able to track issues with prop-types definitions and incrementally resolve those.

### CODE REVIEW

**Comment by kschiffer** [comment]
We should be aware that if we should reintroduce linting to our githooks (which we should do soon), we would see very lengthy warning output for every commit.

**Comment by bafonins** [comment]
> Likewise, these will be output in the CI log for every build.

We can supress `eslint` warnings with the `--quiet` flag. Shall we this until https://github.com/TheThingsNetwork/lorawan-stack/issues/1086 is resolved?

**Comment by kschiffer** [comment]
I think that would be good indeed.

**Comment by bafonins** [comment]
@htdvisser please check `js.go`

File: .mage/js.go [hunk 1]
```diff
@@ -324,7 +324,9 @@ func (js Js) Lint() error {
 	if err != nil {
 		return err
 	}
-	return eslint("./pkg/webui/**/*.js", "./config/**/*.js", "--no-ignore", "--color")
+	// TODO: Remove `--quiet` after https://github.com/TheThingsNetwork/lorawan-stack/issues/1086
```
Comment (rvolosatovs) [inline .mage/js.go]: Usually we make this of form `TODO: <Action> (<issue URL>)`

**Pull request merged**